discovery-dns/src/main/scala/akka/discovery/dns/DnsSimpleServiceDiscovery.scala

```diff
@@ -11,26 +12,48 @@ import akka.pattern.ask
 import scala.concurrent.Future
 import scala.concurrent.duration.FiniteDuration
 import akka.discovery._
+import akka.io.dns.{ AAAARecord, ARecord, DnsProtocol, SRVRecord }
+import akka.io.dns.DnsProtocol.{ Ip, Srv }
+import com.typesafe.config.Config
+
+class DnsDiscoverySettings(config: Config) {
+  val LookupType = config.getString("akka.discovery.akka-dns.lookup-type").toLowerCase() match {
+    case "srv" => Srv
+    case "ip" => Ip()
+  }
+}
 
 /**
  * Looks for A records for a given service.
  */
 class DnsSimpleServiceDiscovery(system: ActorSystem) extends SimpleServiceDiscovery {
+
+  // Required for async dns
+  // AkkaVersion.require("discovery-dns", "2.5.14")
+  require(system.settings.config.getString("akka.io.dns.resolver") == "async-dns",
+    "Akka discovery DNS requires akka.io.dns.resolver to be set to async-dns")
+
   import SimpleServiceDiscovery._
 
   private val log = Logging(system, getClass)
   private val dns = IO(Dns)(system)
+  private val settings = new DnsDiscoverySettings(system.settings.config)
+
   import system.dispatcher
 
   override def lookup(name: String, resolveTimeout: FiniteDuration): Future[Resolved] = {
     def cleanIpString(ipString: String): String =
       if (ipString.startsWith("/")) ipString.tail else ipString
 
-    dns.ask(Dns.Resolve(name))(resolveTimeout) map {
-      case resolved: Dns.Resolved =>
-        log.info("Resolved Dns.Resolved: {}", resolved)
-        val addresses = resolved.ipv4.map { entry ⇒
-          ResolvedTarget(cleanIpString(entry.getHostAddress), port = None)
+    log.debug("Resolving {} type {}", name, settings.LookupType)
```

Comment:
are we sure this should not be `if name starts with "_" => srv lookup`?
that's how it was in the other impl as well